scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -2113,7 +2117,16 @@ class FormatOps(
               if !hasSingleTermStat(t) && isBlockStart(t, nft) =>
             Some(new OptionalBracesRegion {
               def owner = t.parent
-              def splits = Some(getSplitsMaybeBlock(ft, nft, t, true))
+              def splits = Some(
+                getSplitsMaybeBlock(
+                  ft,
+                  nft,
+                  t,
+                  true,
+                  danglingKeyword = style.danglingParentheses.ctrlSite &&
+                    ft.left.is[Token.KwWhile]
```

Comment:
should `while` have its own implementation class, like `if` and `do` do?

Reply:
I would say no, since it seems to work fine currently and this is the only difference between normal block and a while. I would hold off on splitting it until more differences pop up.